src/http-gateways/path-gateway.md

```diff
@@ -567,7 +591,10 @@ The following response types require an explicit opt-in, can only be requested w
 - Raw Block (`?format=raw`)
   - Opaque bytes, see [application/vnd.ipld.raw](https://www.iana.org/assignments/media-types/application/vnd.ipld.raw).
 - CAR (`?format=car`)
-  - Arbitrary DAG as a verifiable CAR file or a stream, see [application/vnd.ipld.car](https://www.iana.org/assignments/media-types/application/vnd.ipld.car).
+  - A CAR file or a stream that contains all blocks required to trustless verify the expressed query, see [application/vnd.ipld.car](https://www.iana.org/assignments/media-types/application/vnd.ipld.car).
```

Comment:
I wonder if it might be reasonable to have an optional feature where one can request a CAR file containing as large a subset of this that can be fetched immediately (for some reasonable definition of immediate, perhaps whatever is in your local store). If one is combining results from multiple gateways this might be a reasonable compromise between number of round trips and the length of any one request.

Reply:
Alternatively, a server-side timeout, to give the gateway an idea of how much effort it should put into collecting all the things. Something like `?timeout=1` implying that the response might be incomplete or, in the case of IPNS, expired.

Reply:
- We have something like  the first idea for blocks. 
  - Requesting `?format=raw` with [`Cache-Control: only-if-cached`](https://specs.ipfs.tech/http-gateways/path-gateway/#only-if-cached) in Kubo should abort if it is not in local cache.  
  - If we want something similar for CARs, caching gets tricky, we don't want to cache partial streams, so it sounds like a new parameter `?format=car&dag-scope=cached` is needed -- not against it, but out of the scope of this IPIP., could be future one.
- As for server-side timeouts, they make sense on paper, but in practice client has to implement  own timeout anyway and the server will abort when client cuts the connection (just to be safe when interacting with random gateways on the internet, often running old/custom software/forks). Probably better to spend time elsewhere.